kola/tests/kubernetes/coreosbasic.go

```diff
@@ -0,0 +1,319 @@
+	if err := runInstallScript(master, controllerInstallScript, ETCD_ENDPOINTS); err != nil {
+		return fmt.Errorf("Installing controller: %v", err)
+	}
+
+	workerScript := strings.Replace(workerInstallScript, "CONTROLLER_ENDPOINT=", "CONTROLLER_ENDPOINT="+CONTROLLER_ENDPOINT, 1)
+	for _, worker := range workers {
+		if err := runInstallScript(worker, workerScript, ETCD_ENDPOINTS); err != nil {
+			return fmt.Errorf("Installing controller: %v", err)
+		}
+	}
+
+	// configure kubectl
+	if err := configureKubectl(master, MASTER_HOST); err != nil {
+		return nil
+	}
+
+	// check that all nodes appear in kubectl
+	f := func() error {
+		if err = nodeCheck(master, workers); err != nil {
+			return err
+		}
+		return nil
+	}
+	if err := util.Retry(8, 10*time.Second, f); err != nil {
+		return err
+	}
+
+	// start nginx pod and curl endpoint
+	if err = nginxCheck(master, workers); err != nil {
+		return err
+	}
+
+	// http://kubernetes.io/v1.0/docs/user-guide/secrets/ Also, ensures
+	// https://github.com/coreos/bugs/issues/447 does not re-occur.
+	if err = secretCheck(master, workers); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func generateMasterTLSAssets(master platform.Machine, MASTER_HOST string, K8S_SERVICE_IP string) error {
+	cnf := strings.Replace(masterCNF, "${K8S_SERVICE_IP}", K8S_SERVICE_IP, -1)
+	cnf = strings.Replace(cnf, "${MASTER_HOST}", MASTER_HOST, -1)
+	in := strings.NewReader(cnf)
+	if err := platform.InstallFile(in, master, "/home/core/openssl.cnf"); err != nil {
+		return err
+	}
+
+	var cmds = []string{
+		// gen master assets
+		"openssl genrsa -out ca-key.pem 2048",
+		`openssl req -x509 -new -nodes -key ca-key.pem -days 10000 -out ca.pem -subj "/CN=kube-ca"`,
+		"openssl genrsa -out apiserver-key.pem 2048",
+		`openssl req -new -key apiserver-key.pem -out apiserver.csr -subj "/CN=kube-apiserver" -config openssl.cnf`,
+		"openssl x509 -req -in apiserver.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out apiserver.pem -days 365 -extensions v3_req -extfile openssl.cnf",
+
+		// gen cluster admin keypair
+		"openssl genrsa -out admin-key.pem 2048",
+		`openssl req -new -key admin-key.pem -out admin.csr -subj "/CN=kube-admin"`,
+		"openssl x509 -req -in admin.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out admin.pem -days 365",
+
+		// move into /etc/kubernetes/ssl
+		"sudo mkdir -p /etc/kubernetes/ssl",
+		"sudo cp /home/core/ca.pem /etc/kubernetes/ssl/ca.pem",
+		"sudo cp /home/core/apiserver.pem /etc/kubernetes/ssl/apiserver.pem",
+		"sudo cp /home/core/apiserver-key.pem /etc/kubernetes/ssl/apiserver-key.pem",
+		"sudo chmod 600 /etc/kubernetes/ssl/*-key.pem",
+		"sudo chown root:root /etc/kubernetes/ssl/*-key.pem",
+	}
+
+	for _, cmd := range cmds {
+		b, err := master.SSH(cmd)
+		if err != nil {
+			return fmt.Errorf("Failed on cmd: %s with error: %s and output %s", cmd, err, b)
+		}
+	}
+	return nil
+}
+
+func generateWorkerTLSAssets(master platform.Machine, workers []platform.Machine) error {
```

Comment:
We could probably use crypto/x509 but it'd be far more verbose and we'd have to copy all the files to the machines anyway.

Since this test follows our install guides somewhat closely, I think its less of a burden to maintain if we just shell out to the same openssl commands.